core/types/transaction.go

```diff
@@ -328,8 +328,9 @@ func (tx *Transaction) Mint() *big.Int {
 // This depends on the chainconfig because gas costs
 // can change over time
 func (tx *Transaction) L1Cost(ctx *L1FeeContext) *big.Int {
-	rlp, _ := rlp.EncodeToBytes(tx.inner)
-	l1GasUsed := calculateL1GasUsed(rlp, ctx.Overhead)
+	var rlp bytes.Buffer
+	tx.EncodeRLP(&rlp)
+	l1GasUsed := calculateL1GasUsed(rlp.Bytes(), ctx.Overhead)
```

Comment:
I think this should be `tx.Data`, not the encoded tx.

Reply:
I was wrong here.

Reply:
Restored.